src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/NuGetVersionComparer.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.UpgradeAssistant.MSBuild
+{
+    public class NuGetVersionComparer : IVersionComparer
+    {
+        public int Compare(string? x, string? y)
+        {
+            if (NuGetVersion.TryParse(x, out var nx))
```

Comment:
It looks like we might be missing a `!` here (and on the next case)

Reply:
Added tests